src/java.base/share/classes/java/io/ObjectInputStream.java

```diff
@@ -357,11 +357,10 @@ private static class Logging {
     private static final ClassByNameCache classByNameCache =
             isClassCachingEnabled ? new ClassByNameCache() : null;
```

Comment:
Suggest the field `isClassCachingEnabled` can also be removed and replaced by `(null != classByNameCache)`.
```java
    /*
     * Unless system property "com.ibm.enableClassCaching" is false, cache LUDCL
     * (Latest User-Defined Class Loader) until completion of read requests.
     */
    private static final ClassByNameCache classByNameCache =
            Boolean.parseBoolean(System.getProperty("com.ibm.enableClassCaching", "true"))
                ? new ClassByNameCache()
                : null;
```